### PR TITLE
Moving directconnect (5) resources to v1beta1 version

### PR DESCRIPTION
**Pull request by svscheg**
### Description of your changes
Moving 5 resources in the `directconnect` group:

- [x] [`aws_dx_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_connection)
- [x] [`aws_dx_connection_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_connection_association)
- [x] [`aws_dx_gateway`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_gateway)
- [x] [`aws_dx_lag`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_lag)
- [x] [`aws_dx_public_virtual_interface`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_public_virtual_interface)

Fixes https://github.com/upbound/provider-aws/issues/12

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually
Uptest
-[`aws_dx_connection`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_connection) - https://github.com/upbound/provider-aws/actions/runs/3602736273/jobs/6070071036
-[`aws_dx_connection_association`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_connection_association) - https://github.com/upbound/provider-aws/actions/runs/3602743895/jobs/6070087879
-[`aws_dx_gateway`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_gateway) - https://github.com/upbound/provider-aws/actions/runs/3602747011/jobs/6070095148
-[`aws_dx_lag`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_lag) - https://github.com/upbound/provider-aws/actions/runs/3602753604/jobs/6070108699
-[`aws_dx_public_virtual_interface`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/dx_public_virtual_interface) - https://github.com/upbound/provider-aws/actions/runs/3603002470/jobs/6070655731

re-run:
https://github.com/upbound/provider-aws/actions/runs/3628552748
https://github.com/upbound/provider-aws/actions/runs/3627924232
https://github.com/upbound/provider-aws/actions/runs/3627925932
https://github.com/upbound/provider-aws/actions/runs/3627922169
https://github.com/upbound/provider-aws/actions/runs/3627883229

### CODE REVIEW

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/connection.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/connectionassociation.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/gateway.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/lag.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/publicvirtualinterface.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/publicvirtualinterface.yaml"

File: config/directconnect/config.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"github.com/upbound/upjet/pkg/config"
+)
+
+// Configure adds configurations for directconnect group.
+func Configure(p *config.Provider) { // nolint:gocyclo
+	p.AddResourceConfigurator("aws_dx_public_virtual_interface", func(r *config.Resource) {
+		r.References["connection_id"] = config.Reference{
+			Type: "github.com/upbound/provider-aws/apis/directconnect/v1beta1.Connection",
+		}
+		r.UseAsync = true
```
Comment (ulucinar) [inline config/directconnect/config.go:17]: The default mode is [async](https://github.com/upbound/upjet/blob/2bd418a5521b5a4a575c0fae7185bd7f60a06560/pkg/config/common.go#L71) in Upjet:
```suggestion
```

File: config/directconnect/config.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+Copyright 2022 Upbound Inc.
+*/
```
[the remark this answers is not in the document]
Comment (svscheg) [reply config/directconnect/config.go]: Done

File: config/directconnect/config.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import (
+	"github.com/upbound/upjet/pkg/config"
+)
+
+// Configure adds configurations for directconnect group.
+func Configure(p *config.Provider) { // nolint:gocyclo
+	p.AddResourceConfigurator("aws_dx_public_virtual_interface", func(r *config.Resource) {
+		r.References["connection_id"] = config.Reference{
+			Type: "github.com/upbound/provider-aws/apis/directconnect/v1beta1.Connection",
```
[the remark this answers is not in the document]
Comment (svscheg) [reply config/directconnect/config.go]: Done

File: config/directconnect/config.go (hunk 1 above)
Comment (svscheg) [reply config/directconnect/config.go]: Done

File: config/externalname.go [hunk 4]
```diff
@@ -1354,6 +1354,19 @@ var ExternalNameConfigs = map[string]config.ExternalName{
...
+	// Direct Connect Gateways can be imported using the gateway id
+	"aws_dx_gateway": config.IdentifierFromProvider,
+	// Direct Connect connections can be imported using the connection id
+	"aws_dx_connection": config.IdentifierFromProvider,
+	// Direct Connect public virtual interfaces can be imported using the vif id
+	"aws_dx_public_virtual_interface": config.IdentifierFromProvider,
+	// No import
+	"aws_dx_connection_association": config.IdentifierFromProvider,
+	// Direct Connect LAGs can be imported using the lag id
+	"aws_dx_lag": config.IdentifierFromProvider,
```
[the remark this answers is not in the document]
Comment (svscheg) [reply config/externalname.go:1377]: Done

File: examples/directconnect/connectionassociation.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    bandwidth: 1Gbps
+    location: PNP51
+    name: example
+    region: us-west-1
+---
+apiVersion: directconnect.aws.upbound.io/v1beta1
+kind: Lag
+metadata:
+  annotations:
+    meta.upbound.io/example-id: directconnect/v1beta1/lag
```
[the remark this answers is not in the document]
Comment (svscheg) [reply examples/directconnect/connectionassociation.yaml]: Done

File: examples/directconnect/publicvirtualinterface.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    routeFilterPrefixes:
+    - 210.52.109.0/24
+    - 175.45.176.0/22
+    vlan: 4094
+---
+apiVersion: directconnect.aws.upbound.io/v1beta1
+kind: Connection
+metadata:
+  annotations:
+    meta.upbound.io/example-id: directconnect/v1beta1/connectionassociation
```
[the remark this answers is not in the document]
Comment (svscheg) [reply examples/directconnect/publicvirtualinterface.yaml]: Done

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/connection.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/connectionassociation.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/lag.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/publicvirtualinterface.yaml"

**Comment by svscheg** [comment]
/test-examples="examples/directconnect/connectionassociation.yaml"

**Review by ulucinar — APPROVED** [review]
Thank you @svscheg, lgtm.

**Review by ezgidemirel — APPROVED** [review]
Thanks @svscheg and @ulucinar !!

**Pull request merged**